File: C/Grace/Grace.c
#include <sys/stat.h>
#include <fcntl.h>
#include <stdio.h>
#include <unistd.h>
// Comment
#define STR "#include <sys/stat.h>%1$c#include <fcntl.h>%1$c#include <stdio.h>%1$c#include <unistd.h>%1$c// Comment%1$c#define STR %2$c%3$s%2$c%1$c#define KID int fd = open(%2$cGrace_kid.c%2$c, O_CREAT | O_WRONLY, 0644);if(fd==-1)return 1;dprintf(fd, STR, 10, 34, STR);close(fd);%1$c#define MAIN int main(){KID}%1$cMAIN"
#define KID int fd = open("Grace_kid.c", O_CREAT | O_WRONLY, 0644);if(fd==-1)return 1;dprintf(fd, STR, 10, 34, STR);close(fd);
#define MAIN int main(){KID}
MAIN

File: Rust/Grace/Grace.rs
use std::fs;
// Comment
macro_rules! STR {
	() => {
		"use std::fs;{0}// Comment{0}macro_rules! STR {{{0}	() => {{{0}		{1}{3}{1}{0}	}};{0}}}{0}macro_rules! KID {{{0}	() => {{{0}		let str = format!(STR!(), '{2}n', '{1}', '{2}{2}', STR!());{0}		fs::write({1}Grace_kid.rs{1}, str)?{0}	}};{0}}}{0}macro_rules! MAIN {{{0}	() => {{{0}		fn main() -> std::io::Result<()> {{{0}			KID!();{0}			Ok(()){0}		}}{0}	}};{0}}}{0}MAIN!();"
	};
}
macro_rules! KID {
	() => {
		let str = format!(STR!(), '\n', '"', '\\', STR!());
		fs::write("Grace_kid.rs", str)?
	};
}
macro_rules! MAIN {
	() => {
		fn main() -> std::io::Result<()> {
			KID!();
			Ok(())
		}
	};
}
MAIN!();